### DISCUSSION TITLE
Hasura MySQL Preview : Admin Console Login Fail

### DISCUSSION THREAD

**Discussion by ReghuShanker** [hasura/graphql-engine#5827] [General]
I am trying to login to the Hasura MySQL preview console. When I enter the admin password, it logins and I see the console for a second. The console then returns back to the login screen. I checked my docker-compose file and verified I am using the correct admin password.  Server is in Digital Ocean. 

The following is the screencast of the issue:  https://nimb.ws/8aOfkK

My docker-compose is the following : 
[docker-compose.txt](https://github.com/hasura/graphql-engine/files/5267758/docker-compose.txt)

When I check the network requests I at first login requests returns success but later another request is denied permission.
Screenshots of the failed requests:

```
Request URL: http://128.199.22.235/v1/query
Request Method: POST
Status Code: 400 Bad Request
Remote Address: 128.199.22.235:80
Referrer Policy: strict-origin-when-cross-origin
Access-Control-Allow-Credentials: true
Access-Control-Allow-Methods: GET,POST,PUT,PATCH,DELETE,OPTIONS
Access-Control-Allow-Origin: http://128.199.22.235
Content-Length: 81
Content-Type: application/json; charset=utf-8
Date: Wed, 23 Sep 2020 11:30:10 GMT
Server: Caddy
Server: Warp/3.3.10
Accept: */*
Accept-Encoding: gzip, deflate
Accept-Language: en-US,en;q=0.9,ml;q=0.8
Connection: keep-alive
Content-Length: 87
Content-Type: text/plain;charset=UTF-8
Cookie: pma_lang=en; pmaUser-1=%7B%22iv%22%3A%22TpaLPvi6tc9paKbBZc7IAw%3D%3D%22%2C%22mac%22%3A%2260d80d4c1f5f78b45f0df115cbb9c63c48160334%22%2C%22payload%22%3A%22h3%5C%2FnYHmw0yF5mZGQbZm4Bw%3D%3D%22%7D; phpMyAdmin=9686c6a5f6211f4ab59d17a88a6335fb
DNT: 1
Host: 128.199.22.235
Origin: http://128.199.22.235
Referer: http://128.199.22.235/console
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.102 Safari/537.36
{type: "run_sql", args: {sql: "SELECT version();", cascade: false, read_only: false}}
args: {sql: "SELECT version();", cascade: false, read_only: false}
type: "run_sql"

```

The Hasura log from docker container for the above request is the following:

graphql-engine_1  | {"type":"http-log","timestamp":"2020-09-23T11:27:54.577+0000","level":"error","detail":{"operation":{"user_vars":{"x-hasura-role":"anonymous"},"error":{"path":"$.args","error":"restricted access : admin only","code":"access-denied"},"request_id":"aa58017e-b26e-4b64-af5c-39db9610f0d4","response_size":81,"query":{"args":{"cascade":false,"check_metadata_consistency":null,"sql":"SELECT version();","read_only":false},"version":1,"type":"run_sql"}},"http_info":{"status":400,"http_version":"HTTP/1.1","url":"/v1/query","ip":"103.134.135.89","method":"POST","content_encoding":null}}}

The following is the log for successful requests:

graphql-engine_1  | {"type":"http-log","timestamp":"2020-09-23T11:27:54.577+0000","level":"info","detail":{"operation":{"query_execution_time":0.183807047,"user_vars":{"x-hasura-role":"admin"},"request_id":"eb3b6a6b-c262-4f99-aa58-f5825d0186a9","response_size":1918,"request_read_time":2.531373e-3},"http_info":{"status":200,"http_version":"HTTP/1.1","url":"/v1/query","ip":"103.134.135.89","method":"POST","content_encoding":"gzip"}}}

Seems like x-hasura-role has changed from admin to anonymous not sure if this the reason or not.

If anyone can take a look at it, then it would be really helpful.

**Comment by keag-dev** [comment]
I am experiencing this exact issue, while attempting to use HASURA_GRAPHQL_ADMIN_SECRET to login.
Is this expected behavior on the MySQL preview or do we need to configure the preview differently than a regular hasura instance?

**Comment by preetjdp** [comment]
Solved this by using a chrome extension to set the `x-hasura-admin-secret` header for the Hasura console website

The extension -> https://chrome.google.com/webstore/detail/modheader/idgpnmonknjnojddfkpgkljpfnnfcklj?hl=en

<img width="1440" alt="Screenshot 2021-06-01 at 7 05 29 PM" src="https://user-images.githubusercontent.com/27439197/120332500-6f581700-c30c-11eb-8984-f6c11f9f8823.png">
